src/server/core.js

```diff
@@ -9,29 +9,42 @@ function initTopics(topics) {
   }, {});
 }
 
+// return a changeset to merge with state
 function vote(state, topicId, voterId) {
+  returnSet = {'topics': {}, 'voters': []}
   const voter = state.voters.find(voter => voter.id === voterId);
   if (!voter) {
-    state.topics[topicId].votes++;
-    state.voters.push({ id: voterId, vote: topicId });
```

Comment:
#push creating a side-effect, also like a directly changing topics object property in line above